src/daft-functions/src/distance/cosine.rs

```diff
@@ -0,0 +1,140 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    datatypes::{Field, Float64Array},
+    schema::Schema,
+    DataType, IntoSeries, Series,
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+use simsimd::SpatialSimilarity;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct CosineDistanceFunction {}
+
+macro_rules! compute_cosine_distance {
```

Comment:
i think this _could_ be done with a generic function, but I was having some difficulties expressing it as such.

Reply:
converted this to generic function instead